Code/Framework/AzFramework/AzFramework/IO/LocalFileIO.cpp

```diff
@@ -829,8 +829,21 @@ namespace AZ
                     {
                         return false;
                     }
-                    replacedAliasPath.Native() = AZ::IO::FixedMaxPathString(AZStd::string_view{ aliasValue });
-                    replacedAliasPath.Native() += postAliasView;
+
+                    // Check if the path is input path does not overlap the output path
+                    if (postAliasView.data() < replacedAliasPath.Native().data()
+                        && postAliasView.data() >= replacedAliasPath.Native().data() + replacedAliasPath.Native().size())
```

Comment:
I'm not sure I follow this logic.

If the `postAliasView.data()` pointer (unchanged part of path) is less than the `replacedAliasPath.Native().data()` (output path) pointer then how could that same pointer ever also be greater than the pointer to the output path + output path size?  Would something like this work?
```suggestion
                    if (postAliasView.data() + postAliasView.size() < replacedAliasPath.Native().data()
                        || postAliasView.data() >= replacedAliasPath.Native().data() + replacedAliasPath.Native().size())
```

Reply:
Actually I inverted the check, but not the `&&` and `||`

Reply:
ah this is breaking my brain.  It looks like `postAliasView` - which is a string view of `pathStrView` which comes from the input `path`, may be a view of a string completely unrelated to the output `replacedAliasPath`.   Like if you called the method this way:
```
                AZ::IO::FixedMaxPath pathWithAlias{"@engroot@/blah"};
                AZ::IO::FixedMaxPath pathResolved;
                AZ::IO::FileIOBase::GetInstance()->ReplaceAlias(pathResolved, AZ::IO::PathView{ pathWithAlias});
```

Is the intention that this method should only be called with a subview of the output path?

Reply:
This function can be used with any input path.
The first block is when the input path isn't a subview of the output path.
The else block is when the input path is a subview of the output path.

Reply:
Thanks!  Makes a lot more sense now